e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js

```diff
@@ -88,4 +77,465 @@ describe('Profile popover', () => {
+        verifyPermissionSubSections('channel_admin', 'public', false);
+
+        cy.findByTestId('channel_admin-private_channel-checkbox').scrollIntoView().should('be.visible').click().click();
+
+        // * Verify that all the sub-checkboxes are disabled.
+        verifyPermissionSubSections('channel_admin', 'private', false);
+
+        cy.findByTestId('team_admin-public_channel-checkbox').scrollIntoView().should('be.visible').click().click();
+
+        // * Verify that all the sub-checkboxes are disabled.
+        verifyPermissionSubSections('team_admin', 'public', false);
+
+        cy.findByTestId('team_admin-private_channel-checkbox').scrollIntoView().should('be.visible').click().click();
+
+        // * Verify that all the sub-checkboxes are disabled.
+        verifyPermissionSubSections('team_admin', 'private', false);
+
+        cy.findByTestId('saveSetting').should('be.visible').click();
+
+        cy.apiLogout();
+
+        // # Login
+        cy.apiLogin(otherUser);
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        verifyAddToChannel(testUser, false);
+
+        // # Promote to channel admin
+        promoteToChannelOrTeamAdmin(otherUser, testChannel.id);
+
+        cy.apiLogout();
+
+        cy.apiLogin(otherUser);
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        verifyAddToChannel(testUser, false);
+
+        // # Promote to team admin
+        promoteToChannelOrTeamAdmin(otherUser, testTeam.id, 'teams');
+
+        cy.apiLogout();
+
+        cy.apiLogin(otherUser);
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+        verifyAddToChannel(testUser, false);
+        cy.apiLogout();
+
+        // login as system admin
+        cy.apiAdminLogin();
+
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        verifyAddToChannel(testUser);
+    });
 });
+
+const verifyPermissionSubSections = (category, publicOrPrivate, checked) => {
+    let classCondition;
+    if (checked) {
+        classCondition = 'have.class';
+    } else {
+        classCondition = 'not.have.class';
+    }
+
+    // # Expand the section
+    cy.get('#' + category + '-' + publicOrPrivate + '_channel > .fa').scrollIntoView().should('be.visible').click();
+
+    if (category !== 'channel_admin') {
+        cy.findByTestId(category + '-' + publicOrPrivate + '_channel-create_' + publicOrPrivate + '_channel-checkbox').should(classCondition, 'checked');
+    }
+    cy.findByTestId(category + '-' + publicOrPrivate + '_channel-manage_' + publicOrPrivate + '_channel_properties-checkbox').should(classCondition, 'checked');
+    cy.findByTestId(category + '-' + publicOrPrivate + '_channel-manage_' + publicOrPrivate + '_channel_members_and_read_groups-checkbox').should(classCondition, 'checked');
+    cy.findByTestId(category + '-' + publicOrPrivate + '_channel-delete_' + publicOrPrivate + '_channel-checkbox').should(classCondition, 'checked');
+};
```

Comment:
Using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) (string in backticks) might improve the readability a bit.